spring/src/main/java/com/gcinterceptor/spring/SpringGcInterceptor.java

```diff
@@ -29,19 +27,24 @@
 	@Override
 	public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
 			throws Exception {
-		ShedResponse shedResponse = gci.before(request.getHeader(GCI_HEADERS_NAME));
-		if (shedResponse.shouldShed) {
-			response.setStatus(HttpStatus.SERVICE_UNAVAILABLE.value());
-			response.setContentLength(0);
+		
+		switch (request.getHeader(GCI_HEADERS_NAME)) {
+		case "":
```

Comment:
Please check the APIs (https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletRequest.html#getHeader(java.lang.String))

This method returns null if the header is not present. It is different from the ruby case :)